src/test/java/io/aiven/kafka/connect/gcs/config/GcsSinkCredentialsConfigTest.java

```diff
@@ -0,0 +1,187 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.MockedStatic;
+
+/**
+ * Tests {@link GcsSinkConfig} class.
+ */
+final class GcsSinkCredentialsConfigTest {
+    @ParameterizedTest
+    @ValueSource(strings = { "", "{{topic}}", "{{partition}}", "{{start_offset}}", "{{topic}}-{{partition}}",
+            "{{topic}}-{{start_offset}}", "{{partition}}-{{start_offset}}",
+            "{{topic}}-{{partition}}-{{start_offset}}-{{unknown}}" })
+    void incorrectFilenameTemplates(final String template) {
+        final Map<String, String> properties = Map.of("file.name.template", template, "gcs.bucket.name", "some-bucket");
+
+        final ConfigValue configValue = GcsSinkConfig.configDef()
+                .validate(properties)
+                .stream()
+                .filter(x -> GcsSinkConfig.FILE_NAME_TEMPLATE_CONFIG.equals(x.name()))
+                .findFirst()
+                .get();
+        assertFalse(configValue.errorMessages().isEmpty());
+
+        final var throwable = assertThrows(ConfigException.class, () -> new GcsSinkConfig(properties));
+        assertTrue(throwable.getMessage().startsWith("Invalid value "));
+    }
+
+    @Test
+    void gcsCredentialsPath() {
+        final Map<String, String> properties = Map.of("gcs.bucket.name", "test-bucket", "gcs.credentials.path",
+                Thread.currentThread().getContextClassLoader().getResource("test_gcs_credentials.json").getPath());
+
+        assertConfigDefValidationPasses(properties);
+
+        final GcsSinkConfig config = new GcsSinkConfig(properties);
+        final UserCredentials credentials = (UserCredentials) config.getCredentials();
+        assertEquals("test-client-id", credentials.getClientId());
+        assertEquals("test-client-secret", credentials.getClientSecret());
+    }
+
+    @Test
+    void gcsCredentialsJson() throws IOException {
+        final Map<String, String> properties = new HashMap<>();
+        properties.put("gcs.bucket.name", "test-bucket");
+
+        final String credentialsJson = Resources.toString(
+                Thread.currentThread().getContextClassLoader().getResource("test_gcs_credentials.json"),
+                StandardCharsets.UTF_8);
+        properties.put("gcs.credentials.json", credentialsJson);
+
+        assertConfigDefValidationPasses(properties);
+
+        final GcsSinkConfig config = new GcsSinkConfig(properties);
+        final UserCredentials credentials = (UserCredentials) config.getCredentials();
+        assertEquals("test-client-id", credentials.getClientId());
+        assertEquals("test-client-secret", credentials.getClientSecret());
+    }
+
+    /**
+     * This test validates that the NoCredentials are used when default is specified as false. This behaviour mimics
+     * that of the Tiered Storage Manager.
```

Comment:
The TS reference here as it doesn't add much without a reference -- we can just omit it I think
```suggestion
     * This test validates that the NoCredentials are used when default is specified as false.
```